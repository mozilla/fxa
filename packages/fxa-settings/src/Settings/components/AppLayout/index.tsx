/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import HeaderLockup from '../HeaderLockup';
import ContentSkip from '../ContentSkip';
import Footer from 'fxa-react/components/Footer';
import { AlertBar } from '../AlertBar';

type AppLayoutProps = {
  children: React.ReactNode;
};

export const AppLayout = ({ children }: AppLayoutProps) => {
  return (
    <div
      className="flex flex-col justify-between min-h-screen"
      data-testid="app"
    >
      <ContentSkip />
      <HeaderLockup />
      <div className="max-w-screen-desktopXl flex-1 w-full mx-auto tablet:px-20 desktop:px-12">
        <main id="main" data-testid="main" className="w-full">
          <AlertBar />
          {children}
        </main>
      </div>
      <Footer />
    </div>
  );
};

export default AppLayout;
