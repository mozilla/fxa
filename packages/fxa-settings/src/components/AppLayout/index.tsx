/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import mozLogo from 'fxa-react/images/moz-logo-bw-rgb.svg';
import LinkExternal from 'fxa-react/components/LinkExternal';
import { useLocalization } from '@fluent/react';
import Head from 'fxa-react/components/Head';
import classNames from 'classnames';

type AppLayoutProps = {
  // TODO: FXA-6803 - the title prop should be made mandatory
  // the string should be localized
  title?: string;
  children: React.ReactNode;
  widthClass?: string;
};

export const AppLayout = ({ title, children, widthClass }: AppLayoutProps) => {
  const { l10n } = useLocalization();
  return (
    <>
      <Head {...{ title }} />
      <div
        className="flex min-h-screen flex-col items-center"
        data-testid="app"
      >
        <div id="body-top" className="w-full hidden mobileLandscape:block" />
        <main className="mobileLandscape:flex mobileLandscape:items-center mobileLandscape:flex-1 pt-14">
          <section>
            <div className={classNames('card', widthClass)}>{children}</div>
          </section>
        </main>

        <footer className="hidden mobileLandscape:block w-full p-8">
          <LinkExternal
            rel="author"
            href="https://www.mozilla.org/about/?utm_source=firefox-accounts&amp;utm_medium=Referral"
          >
            <img
              src={mozLogo}
              alt={l10n.getString(
                'app-footer-mozilla-logo-label',
                null,
                'Mozilla logo'
              )}
              className="w-32"
            />
          </LinkExternal>
        </footer>
      </div>
      <div id="body-bottom" className="w-full block mobileLandscape:hidden" />
    </>
  );
};

export default AppLayout;
