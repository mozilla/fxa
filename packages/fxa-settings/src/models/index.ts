export * from './AppSettingsContext';
export * from './AlertBarInfo';
export * from './Account';
export * from './Session';
export * from './hooks';
