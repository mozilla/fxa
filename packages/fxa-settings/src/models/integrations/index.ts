/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

export * from './base-integration';
export * from './channel-info';
export * from './client-info';
export * from './oauth-web-integration';
export * from './oauth-native-integration';
export * from './pairing-authority-integration';
export * from './pairing-supplicant-integration';
export * from './signin-signup-info';
export * from './supplicant-info';
export * from './sync-basic-integration';
export * from './sync-desktop-v3-integration';
export * from './web-integration';
export * from './third-party-auth-callback-integration';
