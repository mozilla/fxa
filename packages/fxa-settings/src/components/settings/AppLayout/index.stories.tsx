import React from 'react';
import { storiesOf } from '@storybook/react';
import AppLayout from './index';

storiesOf('Components/AppLayout', module).add('basic', () => (
  <AppLayout>
    <p>App contents go here</p>
  </AppLayout>
));
