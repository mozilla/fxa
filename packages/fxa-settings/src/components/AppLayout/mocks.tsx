/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import { RelierCmsInfo } from '../../models/integrations';

export const MOCK_CMS_INFO_VALID_LINEAR_BG = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
    backgrounds: {
      defaultLayout: 'linear-gradient(135deg, #667eea 0%, #764ba2 100%)',
    },
    pageTitle: 'Test App - Custom Title',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_VALID_RADIAL_BG = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
    backgrounds: {
      defaultLayout: 'radial-gradient(circle, #ff6b6b, #4ecdc4)',
    },
    pageTitle: 'Test App - Custom Title',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_NO_BG = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
    pageTitle: 'Test App - Custom Title',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_INVALID_BG_COLOR = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
    backgrounds: {
      defaultLayout: 'solid-color-red',
    },
    pageTitle: 'Test App - Custom Title',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_WITH_PAGE_TITLE = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
    pageTitle: 'CMS Custom Title',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_NO_PAGE_TITLE = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_HEADER_LOGO = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: undefined,
    logoUrl: undefined,
    logoAltText: undefined,
    headerLogoUrl: 'https://example.com/cms-logo.png',
    headerLogoAltText: 'CMS Custom Logo',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_HEADER_LOGO_NO_ALT = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: undefined,
    logoUrl: undefined,
    logoAltText: undefined,
    headerLogoUrl: 'https://example.com/cms-logo.png',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_DEFAULT_LOGO = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: undefined,
    logoAltText: undefined,
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_HEADER_LOGO_WITH_OTHER_PROPS = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    headerLogoUrl: 'https://example.com/cms-logo.png',
    headerLogoAltText: 'CMS Custom Logo',
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/other-logo.png',
    logoAltText: 'Other Logo',
    backgrounds: {
      defaultLayout: 'linear-gradient(135deg, #667eea 0%, #764ba2 100%)',
    },
    pageTitle: 'Test App - Custom Title',
  },
} as RelierCmsInfo;

export const INTEGRATION_WITH_UNDEFINED_GETCMSINFO = {
  getCmsInfo: () => undefined,
};

export const INTEGRATION_WITH_NO_GETCMSINFO = {} as Record<string, never>;

export function createIntegration(cmsInfo?: RelierCmsInfo) {
  return {
    getCmsInfo: () => cmsInfo,
  };
}
