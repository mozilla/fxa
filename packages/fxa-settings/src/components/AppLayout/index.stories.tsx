/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import AppLayout from './index';
import { Meta } from '@storybook/react';
import { withLocalization } from 'fxa-react/lib/storybooks';
import { RelierCmsInfo } from '../../models/integrations';
import { MOCK_CMS_INFO } from '../../pages/mocks';

export default {
  title: 'Components/AppLayout',
  component: AppLayout,
  decorators: [withLocalization],
} as Meta;

export const Basic = () => (
  <AppLayout>
    <h1 className="card-header">Header content</h1>
    <p className="mt-2">Paragraph content here</p>
  </AppLayout>
);

export const WithoutCard = () => (
  <AppLayout wrapInCard={false}>
    <div className="border-2 border-blue-300 p-4 rounded-lg shadow-lg">
      <h1 className="card-header">Header</h1>
      <p className="mt-2">
        The content provides it's own card, instead of being wrapped in one
        provided by the layout.
      </p>
    </div>
  </AppLayout>
);

export const WithTitle = () => (
  <AppLayout title="Custom Page Title">
    <h1 className="card-header">Header content</h1>
    <p className="mt-2">Paragraph content here</p>
  </AppLayout>
);

export const WithWidthClass = () => (
  <AppLayout widthClass="card-xl">
    <h1 className="card-header">Wide content</h1>
    <p className="mt-2">This content uses a wider card layout</p>
  </AppLayout>
);

export const WithIntegration = () => {
  return (
    <AppLayout cmsInfo={MOCK_CMS_INFO}>
      <h1 className="card-header">Header content</h1>
      <p className="mt-2">Paragraph content here</p>
    </AppLayout>
  );
};

export const WithCmsInfoNoBackground = () => {
  const mockCmsInfo = {
    name: 'Test App',
    clientId: 'test123',
    entrypoint: 'test',
    shared: {
      buttonColor: '#0078d4',
      logoUrl: 'https://example.com/logo.png',
      logoAltText: 'Test App Logo',
      pageTitle: 'Test App - Custom Title',
      // No backgroundColor, so no background image should be applied
    },
  } as RelierCmsInfo;

  return (
    <AppLayout cmsInfo={mockCmsInfo}>
      <h1 className="card-header">Header content</h1>
      <p className="mt-2">Paragraph content here</p>
    </AppLayout>
  );
};

export const WithCmsInfoInvalidBackground = () => {
  const mockCmsInfo = {
    name: 'Test App',
    clientId: 'test123',
    entrypoint: 'test',
    shared: {
      buttonColor: '#0078d4',
      logoUrl: 'https://example.com/logo.png',
      logoAltText: 'Test App Logo',
      backgroundColor: 'invalid-color',
      pageTitle: 'Test App - Custom Title',
    },
  } as RelierCmsInfo;

  return (
    <AppLayout cmsInfo={mockCmsInfo}>
      <h1 className="card-header">Header content</h1>
      <p className="mt-2">Paragraph content here</p>
    </AppLayout>
  );
};

export const WithCmsInfoRadialGradient = () => {
  const mockCmsInfo = {
    name: 'Test App',
    clientId: 'test123',
    entrypoint: 'test',
    shared: {
      buttonColor: '#0078d4',
      logoUrl: 'https://example.com/logo.png',
      logoAltText: 'Test App Logo',
      backgroundColor: 'radial-gradient(circle, #ff6b6b, #4ecdc4)',
      pageTitle: 'Test App - Custom Title',
    },
  } as RelierCmsInfo;

  return (
    <AppLayout cmsInfo={mockCmsInfo}>
      <h1 className="card-header">Header content</h1>
      <p className="mt-2">Paragraph content here</p>
    </AppLayout>
  );
};

export const WithCmsInfoUndefined = () => {
  return (
    <AppLayout cmsInfo={undefined}>
      <h1 className="card-header">Header content</h1>
      <p className="mt-2">Paragraph content here</p>
    </AppLayout>
  );
};

export const WithoutCmsInfo = () => {
  return (
    <AppLayout>
      <h1 className="card-header">Header content</h1>
      <p className="mt-2">Paragraph content here</p>
    </AppLayout>
  );
};
