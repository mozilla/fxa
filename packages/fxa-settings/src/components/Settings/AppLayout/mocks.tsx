/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import AppLayout from '.';
import { createMockSettingsIntegration } from '../mocks';

// Does not follow "Subject" naming convention for clarity because we use
// this throughout the app and not isolated tests.
export const MockSettingsAppLayout = ({
  children,
}: {
  children: React.ReactNode;
}) => {
  return (
    <AppLayout integration={createMockSettingsIntegration()}>
      {children}
    </AppLayout>
  );
};
