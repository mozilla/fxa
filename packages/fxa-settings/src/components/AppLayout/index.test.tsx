/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import { screen } from '@testing-library/react';
import { renderWithLocalizationProvider } from 'fxa-react/lib/test-utils/localizationProvider';
import AppLayout from '.';
import { RelierCmsInfo } from '../../models/integrations';

describe('<AppLayout />', () => {
  it('renders as expected with children', async () => {
    renderWithLocalizationProvider(
      <AppLayout>
        <p>Hello, world!</p>
      </AppLayout>
    );

    expect(screen.getByTestId('app')).toBeInTheDocument();
    screen.getByText('Hello, world!');
    screen.getByRole('main');

    const mozLink = screen.getByRole('link');
    expect(mozLink).toHaveAttribute('rel', 'author');
    expect(mozLink).toHaveAttribute(
      'href',
      'https://www.mozilla.org/about/?utm_source=firefox-accounts&utm_medium=Referral'
    );
    screen.getByAltText('Mozilla logo');
  });

  it('renders with title prop', async () => {
    renderWithLocalizationProvider(
      <AppLayout title="Test Page Title">
        <p>Hello, world!</p>
      </AppLayout>
    );

    expect(screen.getByTestId('app')).toBeInTheDocument();
    screen.getByText('Hello, world!');
  });

  it('renders with widthClass prop', async () => {
    renderWithLocalizationProvider(
      <AppLayout widthClass="card-xl">
        <p>Hello, world!</p>
      </AppLayout>
    );

    expect(screen.getByTestId('app')).toBeInTheDocument();
    screen.getByText('Hello, world!');
  });

  it('renders with cmsInfo prop and valid background image', async () => {
    const mockCmsInfo = {
      name: 'Test App',
      clientId: 'test123',
      entrypoint: 'test',
      shared: {
        buttonColor: '#0078d4',
        logoUrl: 'https://example.com/logo.png',
        logoAltText: 'Test App Logo',
        backgroundColor:
          'linear-gradient(135deg, #667eea 0%, #764ba2 100%)',
        pageTitle: 'Test App - Custom Title',
      },
    } as RelierCmsInfo;

    renderWithLocalizationProvider(
      <AppLayout cmsInfo={mockCmsInfo}>
        <p>Hello, world!</p>
      </AppLayout>
    );

    expect(screen.getByTestId('app')).toBeInTheDocument();
    screen.getByText('Hello, world!');

    // Check that the CSS custom property is set
    const appElement = screen.getByTestId('app');
    expect(appElement).toHaveStyle({
      '--cms-bg': 'linear-gradient(135deg, #667eea 0%, #764ba2 100%)',
    });
  });

  it('renders with cmsInfo prop and radial gradient background', async () => {
    const mockCmsInfo = {
      name: 'Test App',
      clientId: 'test123',
      entrypoint: 'test',
      shared: {
        buttonColor: '#0078d4',
        logoUrl: 'https://example.com/logo.png',
        logoAltText: 'Test App Logo',
        backgroundColor: 'radial-gradient(circle, #ff6b6b, #4ecdc4)',
        pageTitle: 'Test App - Custom Title',
      },
    } as RelierCmsInfo;

    renderWithLocalizationProvider(
      <AppLayout cmsInfo={mockCmsInfo}>
        <p>Hello, world!</p>
      </AppLayout>
    );

    expect(screen.getByTestId('app')).toBeInTheDocument();
    screen.getByText('Hello, world!');

    // Check that the CSS custom property is set
    const appElement = screen.getByTestId('app');
    expect(appElement).toHaveStyle({
      '--cms-bg': 'radial-gradient(circle, #ff6b6b, #4ecdc4)',
    });
  });

  it('renders with cmsInfo prop but no background image when backgroundColor is missing', async () => {
    const mockCmsInfo = {
      name: 'Test App',
      clientId: 'test123',
      entrypoint: 'test',
      shared: {
        buttonColor: '#0078d4',
        logoUrl: 'https://example.com/logo.png',
        logoAltText: 'Test App Logo',
        pageTitle: 'Test App - Custom Title',
        // No backgroundColor
      },
    } as RelierCmsInfo;

    renderWithLocalizationProvider(
      <AppLayout cmsInfo={mockCmsInfo}>
        <p>Hello, world!</p>
      </AppLayout>
    );

    expect(screen.getByTestId('app')).toBeInTheDocument();
    screen.getByText('Hello, world!');

    // Check that no CSS custom property is set
    const appElement = screen.getByTestId('app');
    expect(appElement).not.toHaveStyle({ '--cms-bg': expect.any(String) });
  });

  it('renders with cmsInfo prop but no background image when backgroundColor is invalid', async () => {
    const mockCmsInfo = {
      name: 'Test App',
      clientId: 'test123',
      entrypoint: 'test',
      shared: {
        buttonColor: '#0078d4',
        logoUrl: 'https://example.com/logo.png',
        logoAltText: 'Test App Logo',
        backgroundColor: 'solid-color-red', // Not a valid background-image value
        pageTitle: 'Test App - Custom Title',
      },
    } as RelierCmsInfo;

    renderWithLocalizationProvider(
      <AppLayout cmsInfo={mockCmsInfo}>
        <p>Hello, world!</p>
      </AppLayout>
    );

    expect(screen.getByTestId('app')).toBeInTheDocument();
    screen.getByText('Hello, world!');

    // Check that no CSS custom property is set
    const appElement = screen.getByTestId('app');
    expect(appElement).not.toHaveStyle({ '--cms-bg': expect.any(String) });
  });

  it('renders with cmsInfo prop but no background image when cmsInfo is undefined', async () => {
    renderWithLocalizationProvider(
      <AppLayout cmsInfo={undefined}>
        <p>Hello, world!</p>
      </AppLayout>
    );

    expect(screen.getByTestId('app')).toBeInTheDocument();
    screen.getByText('Hello, world!');

    // Check that no CSS custom property is set
    const appElement = screen.getByTestId('app');
    expect(appElement).not.toHaveStyle({ '--cms-bg': expect.any(String) });
  });

  it('renders with cmsInfo prop but no background image when cmsInfo is missing', async () => {
    renderWithLocalizationProvider(
      <AppLayout>
        <p>Hello, world!</p>
      </AppLayout>
    );

    expect(screen.getByTestId('app')).toBeInTheDocument();
    screen.getByText('Hello, world!');

    // Check that no CSS custom property is set
    const appElement = screen.getByTestId('app');
    expect(appElement).not.toHaveStyle({ '--cms-bg': expect.any(String) });
  });

  it('renders with cmsInfo prop and uses CMS page title when available', async () => {
    const mockCmsInfo = {
      name: 'Test App',
      clientId: 'test123',
      entrypoint: 'test',
      shared: {
        buttonColor: '#0078d4',
        logoUrl: 'https://example.com/logo.png',
        logoAltText: 'Test App Logo',
        pageTitle: 'CMS Custom Title',
      },
    } as RelierCmsInfo;

    renderWithLocalizationProvider(
      <AppLayout cmsInfo={mockCmsInfo} title="Default Title">
        <p>Hello, world!</p>
      </AppLayout>
    );

    expect(screen.getByTestId('app')).toBeInTheDocument();
    screen.getByText('Hello, world!');
  });

  it('renders with cmsInfo prop and falls back to default title when CMS page title is not available', async () => {
    const mockCmsInfo = {
      name: 'Test App',
      clientId: 'test123',
      entrypoint: 'test',
      shared: {
        buttonColor: '#0078d4',
        logoUrl: 'https://example.com/logo.png',
        logoAltText: 'Test App Logo',
        // No pageTitle
      },
    } as RelierCmsInfo;

    renderWithLocalizationProvider(
      <AppLayout cmsInfo={mockCmsInfo} title="Default Title">
        <p>Hello, world!</p>
      </AppLayout>
    );

    expect(screen.getByTestId('app')).toBeInTheDocument();
    screen.getByText('Hello, world!');
  });

  it('renders CMS header logo when headerLogoUrl is provided', async () => {
    const mockCmsInfo = {
      name: 'Test App',
      clientId: 'test123',
      entrypoint: 'test',
      shared: {
        headerLogoUrl: 'https://example.com/cms-logo.png',
        headerLogoAltText: 'CMS Custom Logo',
      },
    } as RelierCmsInfo;

    renderWithLocalizationProvider(
      <AppLayout cmsInfo={mockCmsInfo}>
        <p>Hello, world!</p>
      </AppLayout>
    );

    const cmsLogo = screen.getByAltText('CMS Custom Logo');
    expect(cmsLogo).toBeInTheDocument();
    expect(cmsLogo).toHaveAttribute('src', 'https://example.com/cms-logo.png');
    expect(cmsLogo).toHaveClass('h-auto', 'w-[140px]', 'mx-auto', 'mobileLandscape:mx-0');
  });

  it('renders CMS header logo with default alt text when headerLogoAltText is not provided', async () => {
    const mockCmsInfo = {
      name: 'Test App',
      clientId: 'test123',
      entrypoint: 'test',
      shared: {
        headerLogoUrl: 'https://example.com/cms-logo.png',
        // No headerLogoAltText
      },
    } as RelierCmsInfo;

    renderWithLocalizationProvider(
      <AppLayout cmsInfo={mockCmsInfo}>
        <p>Hello, world!</p>
      </AppLayout>
    );

    const cmsLogo = screen.getByAltText('logo');
    expect(cmsLogo).toBeInTheDocument();
    expect(cmsLogo).toHaveAttribute('src', 'https://example.com/cms-logo.png');
  });

  it('renders default Mozilla logo when headerLogoUrl is not provided', async () => {
    const mockCmsInfo = {
      name: 'Test App',
      clientId: 'test123',
      entrypoint: 'test',
      shared: {
        // No headerLogoUrl
        buttonColor: '#0078d4',
      },
    } as RelierCmsInfo;

    renderWithLocalizationProvider(
      <AppLayout cmsInfo={mockCmsInfo}>
        <p>Hello, world!</p>
      </AppLayout>
    );

    const defaultLogo = screen.getByAltText('Mozilla logo');
    expect(defaultLogo).toBeInTheDocument();
    // The default logo should have the mozLogo src (this will be the imported SVG)
    expect(defaultLogo).toHaveAttribute('src');
  });

  it('renders default Mozilla logo when cmsInfo is not provided', async () => {
    renderWithLocalizationProvider(
      <AppLayout>
        <p>Hello, world!</p>
      </AppLayout>
    );

    const defaultLogo = screen.getByAltText('Mozilla logo');
    expect(defaultLogo).toBeInTheDocument();
    expect(defaultLogo).toHaveAttribute('src');
  });

  it('renders default Mozilla logo when cmsInfo is undefined', async () => {
    renderWithLocalizationProvider(
      <AppLayout cmsInfo={undefined}>
        <p>Hello, world!</p>
      </AppLayout>
    );

    const defaultLogo = screen.getByAltText('Mozilla logo');
    expect(defaultLogo).toBeInTheDocument();
    expect(defaultLogo).toHaveAttribute('src');
  });

  it('renders default Mozilla logo when cmsInfo is missing', async () => {
    renderWithLocalizationProvider(
      <AppLayout>
        <p>Hello, world!</p>
      </AppLayout>
    );

    const defaultLogo = screen.getByAltText('Mozilla logo');
    expect(defaultLogo).toBeInTheDocument();
    expect(defaultLogo).toHaveAttribute('src');
  });

  it('renders CMS header logo when both headerLogoUrl and other CMS properties are provided', async () => {
    const mockCmsInfo = {
      name: 'Test App',
      clientId: 'test123',
      entrypoint: 'test',
      shared: {
        headerLogoUrl: 'https://example.com/cms-logo.png',
        headerLogoAltText: 'CMS Custom Logo',
        buttonColor: '#0078d4',
        logoUrl: 'https://example.com/other-logo.png',
        logoAltText: 'Other Logo',
        backgroundColor: 'linear-gradient(135deg, #667eea 0%, #764ba2 100%)',
        pageTitle: 'Test App - Custom Title',
      },
    } as RelierCmsInfo;

    renderWithLocalizationProvider(
      <AppLayout cmsInfo={mockCmsInfo}>
        <p>Hello, world!</p>
      </AppLayout>
    );

    const cmsLogo = screen.getByAltText('CMS Custom Logo');
    expect(cmsLogo).toBeInTheDocument();
    expect(cmsLogo).toHaveAttribute('src', 'https://example.com/cms-logo.png');

    // Verify that only one logo is rendered
    const allImages = screen.getAllByRole('img');
    expect(allImages).toHaveLength(1);
  });
});
