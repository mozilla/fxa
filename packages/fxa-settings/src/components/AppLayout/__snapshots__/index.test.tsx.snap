// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppLayout /> snapshots renders correctly with CMS: background wrapper 1`] = `
<div
  class="flex min-h-screen flex-col items-center tablet:[background:var(--cms-bg)]"
  data-testid="app"
  style="--cms-bg: linear-gradient(135deg, rgba(255, 255, 255, 0.7) 0%, rgba(255, 26, 26, 0.25) 40%, rgba(230, 0, 0, 0.3) 70%, rgba(179, 0, 0, 0.45) 100%);"
/>
`;

exports[`<AppLayout /> snapshots renders correctly with CMS: header background 1`] = `
<header
  class="w-full px-6 py-4 mobileLandscape:py-6 tablet:[background:var(--cms-header-bg)]"
  style="--cms-header-bg: linear-gradient(135deg, #764ba2 0%, #667eea 100%);"
/>
`;

exports[`<AppLayout /> snapshots renders correctly with CMS: header logo 1`] = `
<img
  alt="custom-header-logo"
  class="h-auto w-[140px] mx-0"
  src="https://cdn.accounts.firefox.com/other/firefox-browser-logo.svg"
/>
`;

exports[`<AppLayout /> snapshots renders correctly with CMS: title 1`] = `"MOCK Shared Title | Mozilla accounts"`;

exports[`<AppLayout /> snapshots renders correctly without CMS: background wrapper 1`] = `
<div
  class="flex min-h-screen flex-col items-center"
  data-testid="app"
/>
`;

exports[`<AppLayout /> snapshots renders correctly without CMS: header background 1`] = `
<header
  class="w-full px-6 py-4 mobileLandscape:py-6"
/>
`;

exports[`<AppLayout /> snapshots renders correctly without CMS: header logo 1`] = `
<img
  alt="Mozilla logo"
  class="h-auto w-[140px] mx-0"
  src="moz-logo-bw-rgb.svg"
/>
`;

exports[`<AppLayout /> snapshots renders correctly without CMS: title 1`] = `"Mozilla accounts"`;
